src/main/java/com/snowflake/kafka/connector/config/IcebergConfigValidator.java

```diff
@@ -0,0 +1,43 @@
+package com.snowflake.kafka.connector.config;
+
+import static com.snowflake.kafka.connector.SnowflakeSinkConnectorConfig.*;
+import static com.snowflake.kafka.connector.internal.streaming.IngestionMethodConfig.SNOWPIPE_STREAMING;
+
+import com.google.common.collect.ImmutableMap;
+import com.snowflake.kafka.connector.SnowflakeSinkConnectorConfig;
+import com.snowflake.kafka.connector.internal.streaming.IngestionMethodConfig;
+import com.snowflake.kafka.connector.internal.streaming.StreamingConfigValidator;
+import java.util.HashMap;
+import java.util.Map;
+
+/** Validates dependencies between parameters in Iceberg mode. */
+public class IcebergConfigValidator implements StreamingConfigValidator {
+
+  private static final String NO_SCHEMATIZATION_ERROR_MESSAGE =
+      "Ingestion to Iceberg table requires " + ENABLE_SCHEMATIZATION_CONFIG + " set to true";
+  private static final String INCOMPATIBLE_INGESTION_METHOD =
+      "Ingestion to Iceberg table is supported only for " + SNOWPIPE_STREAMING;
```

Comment:
SNOWPIPE_STREAMING is just an enum, we could just write
```suggestion
      "Ingestion to Iceberg table is supported only for Snowpipe Streaming";
```

Reply:
Done